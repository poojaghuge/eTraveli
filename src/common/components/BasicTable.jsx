import React, { useState, useEffect } from "react";
import {
  Table,
  TableRow,
  TableBody,
  TableCell,
  TableContainer,
  TableHead,
  Paper,
} from "@mui/material";

function BasicTable(props) {
  const [rows, setRows] = useState(props.data);

  const handleClick = (item) => {
    console.log(item);
    props.showDetails(item);
  };

  useEffect(() => {
    setRows(props.data);
  }, [props.data]);

  return (
    <TableContainer component={Paper}>
      <Table sx={{ minWidth: 650 }} aria-label="simple table">
        <TableHead>
          <TableRow>
            <TableCell>Episode</TableCell>
            <TableCell>Title</TableCell>
            <TableCell>Year</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {rows.length > 0 &&
            rows.map((row) => (
              <TableRow key={row.episode_id} onClick={() => handleClick(row)}>
                <TableCell component="th" scope="row" role="episode">
                  EPISODE {row.episode_id}
                </TableCell>
                <TableCell component="th" scope="row">
                  {row.title}
                </TableCell>
                <TableCell component="th" scope="row">
                  {row.release_date}
                </TableCell>
              </TableRow>
            ))}
        </TableBody>
      </Table>
    </TableContainer>
  );
}

export const MemoisedBasicTable = React.memo(BasicTable);
