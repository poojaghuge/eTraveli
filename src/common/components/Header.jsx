import * as React from "react";
import { styled, alpha } from "@mui/material/styles";
import AppBar from "@mui/material/AppBar";
import Box from "@mui/material/Box";
import Toolbar from "@mui/material/Toolbar";

export default function Header() {
  return (
    <Box sx={{ flexGrow: 1 }}>
      <AppBar position="static" color="primary">
        <Toolbar>
          <div>Movie App</div>
        </Toolbar>
      </AppBar>
    </Box>
  );
}
