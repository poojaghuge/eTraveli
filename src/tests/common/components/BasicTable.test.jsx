import { render, fireEvent } from "@testing-library/react";
import React from "react";
import renderer from "react-test-renderer";
import { MemoisedBasicTable } from "../../../common/components/BasicTable";
import dummyJson from "../services/dummy.json";

const data = dummyJson.data.results;
const showDetailsMock = jest.fn();

describe("Test MemoisedBasicTable", () => {
  it("test if matches the DOM Snapshot", () => {
    const domTree = renderer
      .create(<MemoisedBasicTable data={data} showDetails={() => {}} />)
      .toJSON();
    expect(domTree).toMatchSnapshot();
  });

  it("test if data props is passed table cell header should render", () => {
    const { getByText } = render(
      <MemoisedBasicTable data={data} showDetails={showDetailsMock} />
    );
    expect(getByText("Episode")).toBeInTheDocument();
    expect(getByText("Title")).toBeInTheDocument();
    expect(getByText("Year")).toBeInTheDocument();
  });

  it("test if data props is passed table cell body should render", () => {
    const { getByText, getAllByRole } = render(
      <MemoisedBasicTable data={data} showDetails={showDetailsMock} />
    );
    expect(getAllByRole("episode")).toBeTruthy();
    expect(getByText("A New Hope")).toBeInTheDocument();
    expect(getByText("1977-05-25")).toBeInTheDocument();
  });

  it("test showDetails eevent", () => {
    const { getByText, getAllByRole } = render(
      <MemoisedBasicTable data={data} showDetails={showDetailsMock} />
    );

    const row = getByText("A New Hope");
    fireEvent.click(row);
    expect(showDetailsMock).toHaveBeenCalledTimes(1);
  });
});
