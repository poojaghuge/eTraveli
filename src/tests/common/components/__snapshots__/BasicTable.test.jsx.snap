// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test MemoisedBasicTable test if matches the DOM Snapshot 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-11xur9t-MuiPaper-root-MuiTableContainer-root"
>
  <table
    aria-label="simple table"
    className="MuiTable-root css-quj9j6-MuiTable-root"
    role={null}
  >
    <thead
      className="MuiTableHead-root css-15wwp11-MuiTableHead-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
          scope="col"
        >
          Episode
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
          scope="col"
        >
          Title
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
          scope="col"
        >
          Year
        </th>
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root css-apqrd9-MuiTableBody-root"
      role={null}
    >
      <tr
        className="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        onClick={[Function]}
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          role="episode"
          scope="row"
        >
          EPISODE 
          4
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          scope="row"
        >
          A New Hope
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          scope="row"
        >
          1977-05-25
        </th>
      </tr>
      <tr
        className="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        onClick={[Function]}
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          role="episode"
          scope="row"
        >
          EPISODE 
          5
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          scope="row"
        >
          The Empire Strikes Back
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          scope="row"
        >
          1980-05-17
        </th>
      </tr>
      <tr
        className="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        onClick={[Function]}
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          role="episode"
          scope="row"
        >
          EPISODE 
          6
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          scope="row"
        >
          Return of the Jedi
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          scope="row"
        >
          1983-05-25
        </th>
      </tr>
      <tr
        className="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        onClick={[Function]}
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          role="episode"
          scope="row"
        >
          EPISODE 
          1
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          scope="row"
        >
          The Phantom Menace
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          scope="row"
        >
          1999-05-19
        </th>
      </tr>
      <tr
        className="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        onClick={[Function]}
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          role="episode"
          scope="row"
        >
          EPISODE 
          2
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          scope="row"
        >
          Attack of the Clones
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          scope="row"
        >
          2002-05-16
        </th>
      </tr>
      <tr
        className="MuiTableRow-root css-1q1u3t4-MuiTableRow-root"
        onClick={[Function]}
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          role="episode"
          scope="row"
        >
          EPISODE 
          3
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          scope="row"
        >
          Revenge of the Sith
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          scope="row"
        >
          2005-05-19
        </th>
      </tr>
    </tbody>
  </table>
</div>
`;
